site/server/SiteBaker.tsx

```diff
@@ -274,8 +260,27 @@ export class SiteBaker {
                     path !== "404" &&
                     path !== "google8272294305985984"
             )
-        const toRemove = without(existingSlugs, ...postSlugs)
-        for (const slug of toRemove) {
+
+        return without(existingSlugs, ...postSlugsFromDb)
+    }
+
+    // Bake all Wordpress posts, both blog posts and entry pages
+    async bakePosts() {
+        const postsApi = await wpdb.getPosts()
+
+        const postSlugs = []
+        for (const postApi of postsApi) {
+            const post = await wpdb.getFullPost(postApi)
+
+            postSlugs.push(post.slug)
+            await this.bakePost(post)
+        }
+
+        // Maxes out resources (TODO: RCA)
+        // await Promise.all(bakingPosts.map(post => this.bakePost(post)))
+
+        // Delete any previously rendered posts that aren't in the database
```

Comment:
Nit: in addition to your cleanup above, what if we also put this in a separate method? Maybe `deleteStalePosts`? Then have the `bakePosts` caller call this one after bakePosts.

Reply:
right, it is true that the baking of new posts and deletion of stale ones are conceptually two different things. I don't know if I can resolve to breaking the beautiful symmetry of `bakeAll()` though :D

Reply:
for background, I extracted that behaviour into `getPostSlugsToRemove` to make it testable, as this is quite a touchy piece of code, which directly impacts what is made visible from one bake to the next.

Reply:
@breck7 merging this as-is for now, but feel free to resume that conversation.